[Manual Taxes M2] Add siteID attribute to TaxRate model entity

toupper opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #10595 
<!-- Id number of the GitHub issue this PR addresses. -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
With this PR we add the `siteID` attribute to `TaxRate` entity model, missing from the previous PR https://github.com/woocommerce/woocommerce-ios/pull/10597

## Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

Ensure that unit tests pass correctly. You can also make sure that the model migrates correctly by running a version with model 95 (trunk) and then installing this version over.

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->


---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

toupper committed:
Add siteID attribute to TaxRate entity

joshheald reviewed (approved):
Works as expected

I did notice a number of warnings from Core Data about the transformables, but I think those were probably introduced in a previous PR. Are you aware of them?

```
2023-09-04 09:52:12.590580+0100 WooCommerce[60170:25470146] [Default] Error enumerating unfinished transactions for first transaction listener: Error Domain=ASDErrorDomain Code=509 "No active account" UserInfo={NSLocalizedDescription=No active account}
2023-09-04 09:52:13.545163+0100 WooCommerce[60170:25469931] [error] fault: One or more models in this application are using transformable properties with transformer names that are either unset, or set to NSKeyedUnarchiveFromDataTransformerName. Please switch to using "NSSecureUnarchiveFromData" or a subclass of NSSecureUnarchiveFromDataTransformer instead. At some point, Core Data will default to using "NSSecureUnarchiveFromData" when nil is specified, and transformable properties containing classes that do not support NSSecureCoding will become unreadable.
CoreData: fault: One or more models in this application are using transformable properties with transformer names that are either unset, or set to NSKeyedUnarchiveFromDataTransformerName. Please switch to using "NSSecureUnarchiveFromData" or a subclass of NSSecureUnarchiveFromDataTransformer instead. At some point, Core Data will default to using "NSSecureUnarchiveFromData" when nil is specified, and transformable properties containing classes that do not support NSSecureCoding will become unreadable.
2023-09-04 09:52:13.545490+0100 WooCommerce[60170:25469931] [error] CoreData: One or more models in this application are using transformable properties with transformer names that are either unset, or set to NSKeyedUnarchiveFromDataTransformerName. Please switch to using "NSSecureUnarchiveFromData" or a subclass of NSSecureUnarchiveFromDataTransformer instead. At some point, Core Data will default to using "NSSecureUnarchiveFromData" when nil is specified, and transformable properties containing classes that do not support NSSecureCoding will become unreadable.
CoreData: warning: Property 'cities' on Entity 'TaxRate' is using nil or an insecure NSValueTransformer.  Please switch to using "NSSecureUnarchiveFromData" or a subclass of NSSecureUnarchiveFromDataTransformer instead.
CoreData: warning: Property 'postcodes' on Entity 'TaxRate' is using nil or an insecure NSValueTransformer.  Please switch to using "NSSecureUnarchiveFromData" or a subclass of NSSecureUnarchiveFromDataTransformer instead.
CoreData: warning: Property 'cities' on Entity 'TaxRate' is using nil or an insecure NSValueTransformer.  Please switch to using "NSSecureUnarchiveFromData" or a subclass of NSSecureUnarchiveFromDataTransformer instead.
CoreData: warning: Property 'postcodes' on Entity 'TaxRate' is using nil or an insecure NSValueTransformer.  Please switch to using "NSSecureUnarchiveFromData" or a subclass of NSSecureUnarchiveFromDataTransformer instead.
CoreData: warning: Property 'cities' on Entity 'TaxRate' is using nil or an insecure NSValueTransformer.  Please switch to using "NSSecureUnarchiveFromData" or a subclass of NSSecureUnarchiveFromDataTransformer instead.
CoreData: warning: Property 'postcodes' on Entity 'TaxRate' is using nil or an insecure NSValueTransformer.  Please switch to using "NSSecureUnarchiveFromData" or a subclass of NSSecureUnarchiveFromDataTransformer instead.
```

toupper committed:
Add value transformer name to attributes to remove warnings

toupper committed:
Add migration information and tests related to transformables.

toupper commented:
@joshheald Thanks for the review!

> I did notice a number of warnings from Core Data about the transformables, but I think those were probably introduced in a previous PR. Are you aware of them?

Thanks for letting me know, I fixed those in 1c82a6419adaeb66f77fbf450106e92f7565efb0 and added tests in 8905fc7f4104f7e9c974bb2851d660eab7479df3
